src/alire/alire-platforms-common.adb

```diff
@@ -0,0 +1,54 @@
+with AAA.Enum_Tools;
+
+with Alire.OS_Lib.Subprocess;
+
+package body Alire.Platforms.Common is
+
+   Arch_Detected : Boolean       := False;
+   Arch_Value    : Architectures := Architecture_Unknown;
+
+   ---------------------------
+   -- Machine_Hardware_Name --
+   ---------------------------
+
+   function Machine_Hardware_Name return Architectures is
+   begin
+      if Arch_Detected then
+         return Arch_Value;
+      end if;
+
+      Detect : declare
+         function Is_Known_Arch is new AAA.Enum_Tools.Is_Valid (Architectures);
+
+         Output : AAA.Strings.Vector;
+
+         --  uname is part of coreutils, so it should be available in
```

Comment:
At this point msys2 is not mandatory on Windows so it would be better to have an alternative.
Which could be to always return `X86_64`.